src/colorella/colormap.py

```diff
@@ -18,209 +18,303 @@
+            txt = "Input provided {0} is not recognised".format(
                 self.arg)
-            txt += "\n valid name can be found in colormap.colormap_names"
-            txt += "\n supported file extensions are .cpt and .ct"
+            txt += "\n Use mpl:*name* for Matplotlib Colormaps, cc:*name* for Colorcet Colormaps and cl:*name* to open a" \
+                   " Colormap from the Colormap directory"
+
             raise ValueError(txt)
 
     @property
     def name(self):
         """
         Returns attribute name
-        """
-        return self._name
 
-    def __len__(self):
+        Returns
+        ---------
+        attribute name
         """
-        Returns number of colors in the colormap
-        - for Segmented Colormap: Number of Segments
-        - for Listed Colormap: total number of colors
+        return self._mpl_cm.name
+
+    @classmethod
+    def from_file(cls, filepath, name=None, gradient=True):
         """
-        if isinstance(self._mpl_cm, col.LinearSegmentedColormap):
-            return len(self._mpl_cm._segmentdata.get('red'))
-        elif isinstance(self._mpl_cm, col.ListedColormap):
-            return self._mpl_cm.N
+        Function to open colormap objects from .cpt, .ct, .json files
 
-    def __getitem__(self, item):
+        Parameters:
+        ----------
+        filepath: str
+            path and filename of the colormap to be opened
+        name: str, optional
+            name of the Colormap
+
+        Returns
+        ---------
+        ColorMap object
         """
-        Returns the xth color of the colormap
+
+        mpl_cm = None
+        extension = os.path.splitext(filepath)[1]
+        if '.cpt' == extension:
+            filename, cpt_list, cpt_dict = cptfile2dict(filepath)
+            name = name if name is not None else filename
+            if gradient == False:
+                mpl_cm = col.ListedColormap(name=name, colors=cpt_list)
+            if gradient == True:
+                mpl_cm = col.LinearSegmentedColormap(name=name, segmentdata=cpt_dict)
+        elif '.ct' == extension:
+            filename, gdal_list = ctfile2list(filepath)
+            name = name if name is not None else filename
+            if gradient == False:
+                mpl_cm = col.ListedColormap(name=name, colors=gdal_list)
+            if gradient == True:
+                mpl_cm = col.LinearSegmentedColormap.from_list(name=name, colors=gdal_list)
+        elif '.json' == extension:
+            filename, colors, gradient = json2list(filepath)
+            name = name if name is not None else filename
+            if gradient == False:
+                mpl_cm = col.ListedColormap(colors,  name=name)
+            if gradient == True:
+                mpl_cm = col.LinearSegmentedColormap(name=name, segmentdata=colors)
+
+        else:
+            raise ValueError('File extensions is not recognized, supported file extensions are: .cpt, .ct, .json')
```

Comment:
You could shorten this by putting the inner gradient if clause outside and after the outer if clause.